Python/pytime.c

```diff
@@ -1307,8 +1307,17 @@ _PyTime_localtime(time_t t, struct tm *tm)
 
     error = localtime_s(tm, &t);
     if (error != 0) {
+        // There are only 3 cases where an error could occur:
+        //   * tm is NULL
+        //   * &t is NULL (impossible in this usage)
+        //   * it overflows (t < 0 or t > _MAX__TIME64_T)
         errno = error;
-        PyErr_SetFromErrno(PyExc_OSError);
+        if (tm == NULL) {
```

Comment:
This is the case listed by `localtime_s`. I did not find any existing usage that could possibly call `_PyTime_localtime` with `tm = NULL`, but I did not find any documentation for it either. Of course it does not quite make sense to call it with `tm = NULL`, but if someone ran into the issue in the future(for example, did a `_PyTime_localtime(t, tm_ptr)` and forgot to check the pointer), they would get an `OverflowError` instead of an `OSError`, which would be confusing.

So two things we can do:
1. Leave it as it is as a sanity check - the code path is kind of dead as of now, but could benefit in the future.
2. Document that `tm` can't be `NULL` in the comments and/or do an assertion in the function.

I'm fine with either solution.

Reply:
Please add an assertion to ensure that the function is never called with tm=NULL. A runtime check is not needed. Right, private functions are not documented.

Reply:
Done, thanks for the review!